adofai-roulette.py

```diff
@@ -2,74 +2,50 @@
 
 # setup song, difficulty, and creator lists
 
-tempfile = open("songs.txt", encoding="utf-8")
-songlistbad = tempfile.readlines()
-tempfile2 = open("difficulties.txt", encoding="utf-8")
-difficultylistbad = tempfile2.readlines()
-tempfile3 = open("creators.txt", encoding = "utf-8")
-creatorlistbad = tempfile3.readlines()
+def readLinesAndStrip(filename: str) -> list[str]:
+    with open(filename, encoding="utf-8") as file:
+        lines = file.readlines()
 
-songlist = []
-difficultylist = []
-creatorlist = []
+    return [line.strip() for line in lines]
 
-for a in songlistbad:
-    songlist.append(a.strip())
-for a in difficultylistbad:
-    difficultylist.append(a.strip())
-for a in creatorlistbad:
-    creatorlist.append(a.strip())
+
+songs = readLinesAndStrip("songs.txt")
+difficulties = readLinesAndStrip("difficulties.txt")
+creators = readLinesAndStrip("creators.txt")
 
 # variable stuff
 
 print("type 'balls' to give up, type your percentage to continue")
 
-response = 0
-oldpercent = 0

```

Comment:
Instead of tracking the previous percentage, we later on save the response to a variable declared in the for loop, and don't overwrite the previous response/percentage.